src/language.rs

```diff
@@ -39,15 +40,42 @@ impl<L: SynthLanguage> Analysis<L> for SynthAnalysis {
     fn make(egraph: &EGraph<L, Self>, enode: &L) -> Self::Data {
         let get_cvec = |id: &Id| &egraph[*id].data.cvec;
         let get_interval = |id: &Id| &egraph[*id].data.interval;
+        let get_simplest = |i: &Id| &egraph[*i].data.simplest;
+
+        let simplest = if enode.is_var() || enode.is_constant() {
+            let mut rec = RecExpr::<L>::default();
+            rec.add(enode.clone());
+            rec
+        } else {
+            let mut nodes: Vec<L> = vec![];
+            let mut map: HashMap<Id, Id> = HashMap::default();
+            enode.for_each(|id| {
+                if map.get(&id).is_none() {
+                    let s = get_simplest(&id);
+                    let i = nodes.len();
+                    for n in s.as_ref() {
+                        nodes.push(n.clone().map_children(|id| Id::from(usize::from(id) + i)));
```

Comment:
not following what's happening here. The `Id::from(usize::from(id) + i)` part specifically.

Reply:
actually @bksaiki you did this I think originally? what was this for?

Reply:
Hmmm... good question

Reply:
I think it's just constructing a RecExpr, but I don't remember why